### PR TITLE
drivers: wifi: Support to provide raw scan results

### PR DESCRIPTION
**Pull request by ajayparida**
Support to provide raw beacon and probe response
info to user/application.

### CODE REVIEW

File: drivers/wifi/nrf700x/zephyr/src/zephyr_wifi_mgmt.c [hunk 1]
```diff
@@ -796,3 +796,73 @@ int wifi_nrf_set_power_save_timeout(const struct device *dev,
...
+	if (!vif_ctx_zep) {
+		LOG_ERR("%s: vif_ctx_zep is NULL\n", __func__);
+		return;
+	}
+
+	if (!vif_ctx_zep->scan_in_progresss) {
+		return;
+	}
+
+	rpu_ctx_zep = vif_ctx_zep->rpu_ctx_zep;
```
Comment (sachinthegreen) [inline drivers/wifi/nrf700x/zephyr/src/zephyr_wifi_mgmt.c:824]: Please insert blank line between this and the next line.

File: drivers/wifi/nrf700x/zephyr/src/zephyr_wifi_mgmt.c [hunk 2]
```diff
@@ -796,3 +796,73 @@ int wifi_nrf_set_power_save_timeout(const struct device *dev,
...
+	int val = signal;
+
+	vif_ctx_zep = vif_ctx;
+
+	if (!vif_ctx_zep) {
+		LOG_ERR("%s: vif_ctx_zep is NULL\n", __func__);
+		return;
+	}
+
+	if (!vif_ctx_zep->scan_in_progresss) {
```
Comment (sachinthegreen) [inline drivers/wifi/nrf700x/zephyr/src/zephyr_wifi_mgmt.c:820]: We can have a information log message here.

File: drivers/wifi/nrf700x/zephyr/src/zephyr_wifi_mgmt.c [hunk 3]
```diff
@@ -796,3 +796,73 @@ int wifi_nrf_set_power_save_timeout(const struct device *dev,
...
+					sizeof(*bcn_prb_resp_info) +
+					frame_length);
+		memcpy(bcn_prb_resp_info->data,
+		       wifi_nrf_osal_nbuf_data_get(
+			       fmac_dev_ctx->fpriv->opriv,
+			       nwb),
+		       frame_length);
+	}
+
+	bcn_prb_resp_info->rssi = (val / 100);
```
Comment (sachinthegreen) [inline drivers/wifi/nrf700x/zephyr/src/zephyr_wifi_mgmt.c:858]: Please replace 100 by a macro and add a comment as to why this division by 100 is needed

File: drivers/wifi/nrf700x/zephyr/src/zephyr_wifi_mgmt.c [hunk 4]
```diff
@@ -796,3 +796,73 @@ int wifi_nrf_set_power_save_timeout(const struct device *dev,
...
+	bcn_prb_resp_info->frequency = frequency;
+	bcn_prb_resp_info->frame_length = frame_length;
+	
+	wifi_mgmt_raise_raw_scan_result_event(vif_ctx_zep->zep_net_if_ctx,
+					      bcn_prb_resp_info);
+
+	wifi_nrf_osal_mem_free(fmac_dev_ctx->fpriv->opriv,
+			       bcn_prb_resp_info);
+}
+#endif
```
Comment (sachinthegreen) [inline drivers/wifi/nrf700x/zephyr/src/zephyr_wifi_mgmt.c:868]: #endif /* CONFIG_WIFI_FEAT_RAW_SCAN_RESULTS */

File: drivers/wifi/nrf700x/zephyr/src/zephyr_wifi_mgmt.c [hunk 5]
```diff
@@ -796,3 +796,73 @@ int wifi_nrf_set_power_save_timeout(const struct device *dev,
...
+	fmac_dev_ctx = rpu_ctx_zep->rpu_ctx;
+
+	frame_length = wifi_nrf_osal_nbuf_data_size(fmac_dev_ctx->fpriv->opriv,
+						    nwb);
+
+	if (frame_length > CONFIG_WIFI_RAW_SCAN_RESULT_LENGTH) {
+		bcn_prb_resp_info = wifi_nrf_osal_mem_zalloc(
+					fmac_dev_ctx->fpriv->opriv,
+					sizeof(*bcn_prb_resp_info) +
+					CONFIG_WIFI_RAW_SCAN_RESULT_LENGTH);
```
Comment (sachinthegreen) [inline drivers/wifi/nrf700x/zephyr/src/zephyr_wifi_mgmt.c:839]: Handle the error condition if wifi_nrf_osal_mem_zalloc fails.

File: drivers/wifi/nrf700x/zephyr/src/zephyr_wifi_mgmt.c [hunk 6]
```diff
@@ -796,3 +796,73 @@ int wifi_nrf_set_power_save_timeout(const struct device *dev,
...
+		       wifi_nrf_osal_nbuf_data_get(
+			       fmac_dev_ctx->fpriv->opriv,
+			       nwb),
+		       CONFIG_WIFI_RAW_SCAN_RESULT_LENGTH);
+
+	} else {
+		bcn_prb_resp_info = wifi_nrf_osal_mem_zalloc(
+					fmac_dev_ctx->fpriv->opriv,
+					sizeof(*bcn_prb_resp_info) +
+					frame_length);
```
Comment (sachinthegreen) [inline drivers/wifi/nrf700x/zephyr/src/zephyr_wifi_mgmt.c:850]: Handle the error condition if wifi_nrf_osal_mem_zalloc fails.

File: drivers/wifi/nrf700x/zephyr/src/zephyr_wifi_mgmt.c [hunk 7]
```diff
@@ -796,3 +796,73 @@ int wifi_nrf_set_power_save_timeout(const struct device *dev,
...
+
+	frame_length = wifi_nrf_osal_nbuf_data_size(fmac_dev_ctx->fpriv->opriv,
+						    nwb);
+
+	if (frame_length > CONFIG_WIFI_RAW_SCAN_RESULT_LENGTH) {
+		bcn_prb_resp_info = wifi_nrf_osal_mem_zalloc(
+					fmac_dev_ctx->fpriv->opriv,
+					sizeof(*bcn_prb_resp_info) +
+					CONFIG_WIFI_RAW_SCAN_RESULT_LENGTH);
+		memcpy(bcn_prb_resp_info->data,
```
Comment (sachinthegreen) [inline drivers/wifi/nrf700x/zephyr/src/zephyr_wifi_mgmt.c:840]: Shouldn't an OSAL version of memcpy be used here?

File: drivers/wifi/nrf700x/zephyr/src/zephyr_wifi_mgmt.c [hunk 8]
```diff
@@ -796,3 +796,73 @@ int wifi_nrf_set_power_save_timeout(const struct device *dev,
...
+			       fmac_dev_ctx->fpriv->opriv,
+			       nwb),
+		       CONFIG_WIFI_RAW_SCAN_RESULT_LENGTH);
+
+	} else {
+		bcn_prb_resp_info = wifi_nrf_osal_mem_zalloc(
+					fmac_dev_ctx->fpriv->opriv,
+					sizeof(*bcn_prb_resp_info) +
+					frame_length);
+		memcpy(bcn_prb_resp_info->data,
```
Comment (sachinthegreen) [inline drivers/wifi/nrf700x/zephyr/src/zephyr_wifi_mgmt.c:851]: Shouldn't an OSAL version of memcpy be used here?

**Review by sachinthegreen — CHANGES_REQUESTED** [review]

File: drivers/wifi/nrf700x/zephyr/src/zephyr_wifi_mgmt.c (hunk 2 above)
Comment (ajayparida) [reply drivers/wifi/nrf700x/zephyr/src/zephyr_wifi_mgmt.c]: In case it's associated it will keep on printing the message. 
Kept under commented code.

**Comment by sachinthegreen** [comment]
> @udaynordic

You can add reviewers directly to the reviews list rather than just tagging them.

**Review by udaynordic — APPROVED** [review]

**Review by krish2718 — APPROVED** [review]

**Comment by rlubos** [comment]
I suggest we update SHA once (in https://github.com/nrfconnect/sdk-nrf/pull/11075) and then rebase this PR to speed up the overall merging process.

**Pull request closed without merging**